test/distributed/launcher/bin/test_script_init_method.py

```diff
@@ -72,6 +72,7 @@ def main():
             f"Wrong world size derived. Expected: {world_size}, Got: {derived_world_size}"
         )
 
+    dist.destroy_process_group()
```

Comment:
Do we need add a sleep to reduce flakes? I know some backends like NCCL will not actually destroy immeaditely and willl still be "destroying" when a new process group is initialized. See: https://github.com/pytorch/pytorch/issues/119196

Reply:
Thanks for your reminder. I think the scene in #119196 is about multiple workers? Currently the world_size in test_init_method_env and test_init_method_tcp is 1, but I added a sleep and tested 100 times to ensure safety. @Skylion007